kairos-prover/kairos-circuit-logic/src/transactions.rs

```diff
@@ -0,0 +1,541 @@
+                            .map(|batch| {
+                                batch
+                                    .into_iter()
+                                    .map(|(kind, (sender, recipient, amount))| match kind {
+                                        0 => {
+                                            let (txn, res) = accounts_state
+                                                .random_transfer(sender, recipient, amount);
+                                            (KairosTransaction::Transfer(txn), res)
+                                        }
+                                        1 => {
+                                            let (txn, res) =
+                                                accounts_state.random_withdraw(sender, amount);
+                                            (KairosTransaction::Withdraw(txn), res)
+                                        }
+                                        2 => {
+                                            let (txn, res) = accounts_state
+                                                .random_deposit(sender, recipient, amount);
+                                            (KairosTransaction::Deposit(txn), res)
+                                        }
+                                        _ => unreachable!(),
+                                    })
+                                    .collect::<Vec<_>>()
+                            })
+                            .collect::<Vec<_>>();
+
+                        let (initial_trie, trie_db) = initial_state.build_trie();
+                        Self {
+                            batches,
+                            initial_trie,
+                            trie_db,
+                            initial_state: initial_state.clone(),
+                            final_state: accounts_state,
+                        }
+                    })
+                    .boxed()
+                })
+                .boxed()
+        }
+    }
+
+    /// A test model for the state of the accounts on both L1 and L2.
+    /// This is used to generate random valid transactions.
+    #[derive(Clone, Arbitrary)]
+    #[arbitrary(args = RandomBatchesConfig)]
+    pub struct AccountsState {
+        #[any(args.deref().clone())]
+        pub l1: Accounts<u64>,
+        #[any(args.deref().clone())]
+        pub l2: Accounts<Account>,
+    }
+
+    impl fmt::Debug for AccountsState {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            writeln!(
+                f,
+                "\nAccountsState: {{\nAccountsState.l1.accounts: {:?}",
+                self.l1.accounts
+            )?;
+
+            writeln!(f, "AccountsState.l2.accounts: {:?}\n}}", self.l2.accounts)
+        }
+    }
+
+    impl Default for AccountsState {
+        fn default() -> Self {
+            unreachable!("AccountsState should always be created with AccountsState::new()");
+        }
+    }
```

Comment:
Never seen it before, but it successfully caught usage of proptest::any instead of any_with.

Proptest calls default if you don't provide a the argument.

I don't think I use AccountsState as a Arbitrary parameter any more but it's easy to accidentally do when writing proptests with the test-strategy macros.